glossary/upgrade.md

```diff
@@ -0,0 +1,18 @@
++++
+title = "Upgrade Proposal"
+
+template = "doc.html"
+[extra]
+category = "azimuth"
++++
+
+An **upgrade proposal** is a type of [vote](../voting) action taken by the
+[Galactic Senate](../senate) over [Azimuth](../azimuth). By passing an upgrade
+proposal, [Galaxies](../galaxy) in the Senate agree to change the
+[Ecliptic](../ecliptic) smart contracts.
+
+An upgrade proposal is always a Yes/No vote on whether to adopt a particular set
+of changes to the Ecliptic contracts. The vote concludes at the end of 30 days,
+or whenever an absolute majority is reached, whichever comes first. At the end
+of 30 days the option that has received the most votes is chosen. In either
+case, the upgrade (or lack thereof) is performed automatically.
```

Comment:
Nothing on Ethereum happens automatically. In the case of absolute majority, the vote transaction that pushes it over the line triggers the upgrade. For the "majority + time-up" case, an explicit "update vote status/check result" transaction will need to be made. Notably, anyone can do the latter at any time. Any effects from doing so will only happen if there was a majority vote in favor.

Also these mechanics are consistent across upgrade and document proposals, so may or may not want to be lifted out onto the higher-level ("voting"?) page.

Reply:
https://github.com/urbit/docs/pull/1082/commits/3f741f7556b1ca430e3cf6b12d30702ea49edb3d